### DISCUSSION TITLE
GCP 3 Node cluster install failure - missing DNS records

### DISCUSSION THREAD

**Discussion by lindy63** [okd-project/okd#1551] [General]
**Describe the bug**
GCP 3 Node cluster install fails with main error:
`ERROR Cluster initialization failed because one or more operators are not functioning properly.`
`ERROR failed to initialize the cluster: Cluster operators authentication, console, image-registry, ingress, storage are not available"`

Using the install guide - https://docs.okd.io/latest/installing/installing_gcp/installing-gcp-three-node.html
Note, that I have attempted the install multiple times, and always get the same issue

The Ingress operator reports:
`The "default" ingress controller reports Available=False: IngressControllerUnavailable: One or more status conditions indicate unavailable: DNSReady=False (NoZones: The record isn't present in any zones.)`

The authentication operator reports:
`OAuthServerRouteEndpointAccessibleControllerAvailable: Get "https://oauth-openshift.apps.okd.okd..<mydomain>com/healthz": dial tcp: lookup oauth-openshift.apps.okd.okd.<mydomain>.com on 172.30.0.10:53: no such host (this is likely result of malfunctioning DNS server)`

I had a pre-created authoritative Public GCP Cloud DNS zone, which the OKD installer has addded records to successfully, and also it has created a GCP Private Cloud DNS zone, and which has also been populated OK with a bunch of DNS records

All pods are either running or completed, except for these:
![image](https://user-images.githubusercontent.com/61848016/228920659-ca6dc26a-1611-449b-ad4d-5832db1a880a.png)

which cannot find their respective secrets:
_cloud-network-config-controller-bcc4f7dcc-sxlzh_
`MountVolume.SetUp failed for volume "cloud-provider-secret" : secret "cloud-credentials" not found`
_gcp-pd-csi-driver-controller-7dcc6b4b79-jmmsn_
`MountVolume.SetUp failed for volume "cloud-sa-volume" : secret "gcp-pd-cloud-credentials" not found`

I am suspecting it is DNS related, per the above messages, but if I could get a hint that I am in the right ballpark, that would be most appreciated

**Version**
GCP - 4.12.0-0.okd-2023-03-05-022504

**How reproducible**
100% 

**Log bundle**

**Comment by vrutkovs** [comment]
Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Comment by lindy63** [comment]
Thanks @vrutkovs ! 

Have collected them using `oc adm must-gather`, as my bootstrap VM has been deleted by the installer..

I've replaced my username, domain name & gcp project name in the logs.. anything else identifiable in there that I should worry about before uploading here?

**Comment by vrutkovs** [comment]
I don't think it would have anything identifiable as far as I know

**Comment by lindy63** [comment]
Thanks @vrutkovs .. here you go  [okd_logs.tar.gz](https://github.com/okd-project/okd/files/11122677/okd_logs.tar.gz)

I've also just tried this in another GCP project, in another GCP organisation and got the same error FYI..

Also, if anyone is interested in trying this in GCP, their new C3 instances are currently free in Public Preview! (quotas will apply though, one of my projects had 32 x vCPU quota for C3 so was fine for OKD, but the other one only had 8 and the request to increase it was denied)

![image](https://user-images.githubusercontent.com/61848016/229144700-28f2425d-c6d4-46ba-9a3e-15cc7c5ee8be.png)

**Comment by lindy63** [comment]
Also to mention that on the 2nd project, I used the latest version - 4.12.0-0.okd-2023-03-18-084815

**Comment by vrutkovs** [comment]
You're setting credentials management mode to Manual in install-config, so you'd need to follow https://docs.openshift.com/container-platform/4.12/installing/installing_gcp/manually-creating-iam-gcp.html#manually-creating-iam-gcp before starting the installation

**Reply by lindy63** [reply]
Ah Ok, I wil do that.. 

FYI, in case it helps, i am running the installer from a jumpbox in GCP that uses a service account, as I can't run the installer from my laptop as it gets blocked by my security software :-(

**Comment by MarcoPereira7** [comment]
I am having the exact same issue while trying to install a 3 Node cluster , them being worker and master at the same time, since when i try to create 3 worker nodes they dont get created for some reason...

I am following the procedure that @vrutkovs posted: https://docs.okd.io/latest/installing/installing_gcp/manually-creating-iam-gcp.html#manually-create-iam_manually-creating-iam-gcp since i am also using **credentialsMode: Manual** but i am stuck on step 6. I need to create a secret in <install dir>/manifests for every CredentialsRequest object that was created. But the secret object example has azure secret fields. Does someone here have an example for GCP?

**Reply by lindy63** [reply]
I am at the same place @MarcoPereira7 :-), and I also was confused by that bit., so would be very interested in a GCP example :-)

I instead went for the Workload Identity example, detailed [here](https://docs.openshift.com/container-platform/4.12/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#gcp-workload-identity-mode-installing)

But it failed the first attempt (it didn't even try to use the Service Accounts according to the GCP logs, so got stuck at the same missing DNS records) , so just trying again now (logs to follow soon!) in case I messed something up (normally the case)!

**Reply by lindy63** [reply]
Workload Identity worked for me, thanks very much for the pointer @vrutkovs !!

It failed on my original version `4.12.0-0.okd-2023-03-05-022504` (quite a few times!), but then I tried on the (now) two newer ones and it worked fine, so I must have been doing something wrong with the Managed Identity stuff

Both worked:
[4.12.0-0.okd-2023-04-01-051724](https://github.com/okd-project/okd/releases/tag/4.12.0-0.okd-2023-04-01-051724)
[4.12.0-0.okd-2023-03-18-084815](https://github.com/okd-project/okd/releases/tag/4.12.0-0.okd-2023-03-18-084815)

Also just to check that this repo is correct? - `quay.io/openshift-release-dev/ocp-release:4.12.0-x86_64` as this was the only thing I could find that related to  `ocp-release` for this command

`oc adm release extract --credentials-requests --cloud=gcp --to=./credrequests quay.io/openshift-release-dev/ocp-release:4.12.0-x86_64`
